modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/DXPActivationKeysTable/Header/index.js

```diff
@@ -0,0 +1,185 @@
+			...restSelectedKeys
+		] = selectedKeys.map((key) =>
+			activationKeys.find((activationKey) => activationKey.id === key)
+		);
+
+		const keyCanBeDownloaded = restSelectedKeys.every(
+			(selectedKey) =>
+				DOWNLOADABLE_LICENSE_KEYS.above71(
+					firstSelectedKey,
+					selectedKey
+				) ||
+				DOWNLOADABLE_LICENSE_KEYS.below71(firstSelectedKey, selectedKey)
+		);
+
+		return keyCanBeDownloaded;
+	}, [activationKeys, selectedKeys]);
+
+	const handleAlertStatus = (hasSuccessfullyDownloadedKeys) => {
+		setActivationKeysDownloadStatus(
+			hasSuccessfullyDownloadedKeys
+				? ALERT_DOWNLOAD_TYPE.success
+				: ALERT_DOWNLOAD_TYPE.danger
+		);
+	};
+
+	const ACTIVATION_KEYS_ACTION_ITEMS = getActivationKeysActionsItems(
+		accountKey,
+		licenseKeyDownloadURL,
+		sessionId,
+		handleAlertStatus
+	);
+
+	const ACTIVATION_KEYS_DOWNLOAD_ITEMS = getActivationKeysDownloadItems(
+		isAbleToDownloadAggregateKeys,
+		selectedKeysIDs,
+		licenseKeyDownloadURL,
+		sessionId,
+		handleAlertStatus
+	);
+
+	const getCurrentButton = () => {
+		if (selectedKeys.length > 1) {
+			return (
+				<ButtonDropDown
+					items={ACTIVATION_KEYS_DOWNLOAD_ITEMS}
+					label="Download"
+					menuElementAttrs={{
+						className: 'p-0',
+					}}
+				/>
+			);
+		}
+
+		if (selectedKeys.length) {
+			return (
+				<Button
+					className="btn btn-secondary"
+					onClick={async () =>
+						getActivationKeyDownload(
+							selectedKeys,
+							licenseKeyDownloadURL,
+							sessionId,
+							handleAlertStatus
+						)
+					}
+				>
+					Download
+				</Button>
+			);
+		}
+
+		return (
+			<ButtonDropDown
+				items={ACTIVATION_KEYS_ACTION_ITEMS}
+				label="Actions"
+				menuElementAttrs={{
+					className: 'p-0',
+				}}
+			/>
+		);
+	};
+
+	return (
+		<div>
+			<div className="align-items-center bg-neutral-1 d-flex mb-2 p-3 rounded">
+				{!!selectedKeys.length && (
+					<>
+						<p className="font-weight-semi-bold m-0 ml-auto text-neutral-10">
+							{`${selectedKeys.length} Keys Selected`}
+						</p>
+
+						<Button className="btn-outline-danger cp-deactivate-button mx-2">
+							Deactivate
+						</Button>
+					</>
+				)}
+
+				<div
+					className={classNames({
+						'ml-auto': !selectedKeys.length,
+					})}
+				>
+					{getCurrentButton()}
+				</div>
+			</div>
+
+			{activationKeysDownloadStatus && (
+				<ClayAlert.ToastContainer>
+					<ClayAlert
+						autoClose={
+							AUTO_CLOSE_DOWNLOAD_ALERT_TIME[
+								activationKeysDownloadStatus
+							]
+						}
+						className="cp-activation-key-download-alert"
+						displayType={
+							ALERT_DOWNLOAD_TYPE[activationKeysDownloadStatus]
+						}
+						onClose={() => setActivationKeysDownloadStatus('')}
+					>
+						{
+							ALERT_ACTIVATION_AGGREGATED_KEYS_DOWNLOAD_TEXT[
+								activationKeysDownloadStatus
+							]
+						}
+					</ClayAlert>
+				</ClayAlert.ToastContainer>
+			)}
+
+			{!isAbleToDownloadAggregateKeys && (
+				<ClayAlert displayType="info" title="">
```

Comment:
Just to check, we should have an empty title here? I didnt get it.

Reply:
Nice catch!